spec/factories/patients.rb

```diff
@@ -121,3 +122,31 @@ def updated_patient_payload(existing_patient)
     )]
   )
 end
+
+def create_visit(patient, facility: patient.registration_facility, user: patient.registration_user, visited_at: Time.now)
+  patient.prescription_drugs.where("device_created_at < ?", visited_at).update_all(is_deleted: true)
+  create(:blood_pressure, :with_encounter, :critical, recorded_at: visited_at, facility: facility, patient: patient, user: user)
+  create(:blood_sugar, :fasting, :with_encounter, recorded_at: visited_at, facility: facility, patient: patient, user: user)
+  create_list(:prescription_drug, 2, :protocol, device_created_at: visited_at, facility: facility, patient: patient, user: user)
+  create_list(:prescription_drug, 2, device_created_at: visited_at, facility: facility, patient: patient, user: user)
+  patient.appointments.where("device_created_at < ?", visited_at).update_all(status: :visited)
+  create(:appointment, status: :scheduled, device_created_at: visited_at, scheduled_date: 1.month.after(visited_at), creation_facility: facility, facility: facility, patient: patient, user: user)
+  create(:teleconsultation, patient: patient, facility: facility, requester: user, medical_officer: user, requested_medical_officer: user)
+end
+
+def add_some_visits(patient, visit_count, facility: patient.registration_facility, user: patient.registration_user)
```

Comment:
I think we can remove `some` from the method name 😅

Reply:
If we wanna take inspiration from some of the factorybot patterns, like `create_list`, we could update the method signature to accept the count first, and make everything else a named arg:

`add_visits(visit_count, patient:, facility:, user:)`